### ISSUE TITLE
Build fails with customized cargo target dir

### ISSUE THREAD

**Issue by ksqsf** [remacs/remacs#1543]
I changed the default target dir in `~/.cargo/config`
```
[build]
target-dir = "~/.cache/cargo"
```

remacs seems to use it regardless of this setting.
```
  CCLD     etags
clang: error: no such file or directory: '../rust_src/target/release/libremacs_lib.a'
```

**Comment by ksqsf** [comment]
To fix it:

configure.ac
```
LDFLAGS_REMACS="-L`(cd rust_src && cargo metadata --format-version=1 | jq -r .target_directory)`/\${CARGO_BUILD_DIR}"
```

lib-src/Makefile.in
```
RUST_ARCHIVE = `(cd ../rust_src && cargo metadata --format-version=1 | jq -r .target_directory)`/$(CARGO_BUILD_DIR)/$(REMACSLIB_NAME)
```

This depends on `jq`.

**Comment by shaleh** [comment]
Can you link to some upstream docs? This seems like a lot to ask every project to bend to the user's settings. Shouldn't cargo handle this for us?

**Comment by ksqsf** [comment]
My comment above uses `cargo metadata` which is handled by cargo itself, so this should be the expected solution. I don't know how to reliably and portably extract the `target_directory` field from the returned json though.

**Comment by shaleh** [comment]
Right, what I am asking is why does our project need to change to handle this? Shouldn't cargo know where to write files based on its own config? Are we doing something out of the ordinary?

**Comment by ksqsf** [comment]
* https://github.com/remacs/remacs/blob/master/configure.ac#L5120
* https://github.com/remacs/remacs/blob/master/lib-src/Makefile.in#L196

The target dir is hardcoded to be toplevel/rust_src/target. This leads to the very compilation error I encountered.

**Comment by shaleh** [comment]
Got it. Thank you.
